scripts/data_validation.py

```diff
@@ -0,0 +1,674 @@
+import os
+import pypsa
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+import pycountry
+import logging
+import warnings
+warnings.filterwarnings("ignore")
+
+
+def load_ember_data():
+    """
+    Load the Ember data from a CSV file located in the data folder.
+
+    Returns:
+        pd.DataFrame: Ember data in long format.
+    """
+    ember_data_path = os.path.join(
+        os.getcwd(), "data/validation/ember_yearly_full_release_long_format.csv")
+    ember_data = pd.read_csv(ember_data_path)
+    return ember_data
+
+
+def get_network_path(scenario_folder):
+    """
+    Get the full path to the PyPSA network file from the specified scenario folder.
+    Assumes that only one network file exists in the folder.
+
+    Args:
+        scenario_folder (str): Folder containing the scenario data.
+
+    Returns:
+        str: Full path to the network file.
+
+    Raises:
+        AssertionError: If more than one network file exists in the folder.
+    """
+    results_dir = os.path.join(
+        os.getcwd(), f"workflow/pypsa-earth/results/{scenario_folder}/networks")
+    filenames = os.listdir(results_dir)
+
+    # Ensure only one network file exists
+    if len(filenames) == 1:
+        logging.warning(f"Only 1 network per scenario is allowed currently!")
+    filepath = os.path.join(results_dir, filenames[0])
+
+    return filepath
+
+
+def create_results_dir():
+    """
+    Create a results directory if it does not already exist.
+    This is where output data will be stored.
+    """
+    # Create the folder if it doesn't exist
+    if not os.path.exists("results/plots"):
+        os.makedirs("results/plots")
+
+
+def convert_two_country_code_to_three(country_code):
```

Comment:
I thought to move this function into `_helpers.py`, too. But I think we can keep it here. Because three letter codes will be used only for validation stage.